src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IVariableDeclaration.cs

```diff
@@ -1492,6 +2282,130 @@ static void Main(string[] args)
+	  Declarators:
+	      IVariableDeclaratorOperation (Symbol: System.Int32[] x) (OperationKind.VariableDeclarator, Type: null, IsInvalid) (Syntax: 'x = new int[0]')
+	        Initializer:
+	          IVariableInitializerOperation (OperationKind.VariableInitializer, Type: null, IsInvalid) (Syntax: '= new int[0]')
+	            IArrayCreationOperation (OperationKind.ArrayCreation, Type: System.Int32[], IsInvalid) (Syntax: 'new int[0]')
+	              Dimension Sizes(1):
+	                  ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 0, IsInvalid) (Syntax: '0')
+	              Initializer:
+	                null
+	  Initializer:
+	    null
+";
+
+            var expectedDiagnostics = new[]
+            {
+                // file.cs(6,13): warning CS0219: The variable 'y' is assigned but its value is never used
+                //         int y = 10;
+                Diagnostic(ErrorCode.WRN_UnreferencedVarAssg, "y").WithArguments("y").WithLocation(6, 13),
+                // file.cs(7,25): error CS1674: 'int[]': type used in a using statement must be implicitly convertible to 'System.IDisposable' or implement a suitable 'Dispose' method.
+                //        using( /*<bind>*/int[y switch { int z => 42 }] x = new int[0]/*</bind>*/);
+                Diagnostic(ErrorCode.ERR_NoConvToIDisp, "int[y switch { int z => 42 }] x = new int[0]").WithArguments("int[]").WithLocation(7, 25),
+                // file.cs(7,28): error CS0270: Array size cannot be specified in a variable declaration (try initializing with a 'new' expression)
+                //        using( /*<bind>*/int[y switch { int z => 42 }] x = new int[0]/*</bind>*/);
+                Diagnostic(ErrorCode.ERR_ArraySizeInDeclaration, "[y switch { int z => 42 }]").WithLocation(7, 28),
+                // file.cs(7,29): error CS8652: The feature 'recursive patterns' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //        using( /*<bind>*/int[y switch { int z => 42 }] x = new int[0]/*</bind>*/);
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "y switch { int z => 42 }").WithArguments("recursive patterns").WithLocation(7, 29),
+                // file.cs(7,81): warning CS0642: Possible mistaken empty statement
+                //        using( /*<bind>*/int[y switch { int z => 42 }] x = new int[0]/*</bind>*/);
+                Diagnostic(ErrorCode.WRN_PossibleMistakenNullStatement, ";").WithLocation(7, 81)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<VariableDeclarationSyntax>(new[] { syntaxTree }, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void UsingDeclaration_InvalidIgnoredDimensions_SwitchExpression()
+        {
+            string source = @"
+class C
+{
+    void M1()
+    {
+        int y = 10;
+       using /*<bind>*/int[y switch { int z => 42 }] x = new int[0]/*</bind>*/;
```

Comment:
>using [](start = 7, length = 5)

It doesn't look like we have a test covering ```using``` block.

Reply:
> It doesn't look like we have a test covering using block.
Added.